python/grass/jupyter/timeseries.py

```diff
@@ -0,0 +1,274 @@
+# MODULE:    grass.jupyter.timeseries
+#
+# AUTHOR(S): Caitlin Haedrich <caitlin DOT haedrich AT gmail>
+#
+# PURPOSE:   This module contains functions for visualizing raster and vector
+#            space-time datasets in Jupyter Notebooks
+#
+# COPYRIGHT: (C) 2021 Caitlin Haedrich, and by the GRASS Development Team
+#
+#           This program is free software under the GNU General Public
+#           License (>=v2). Read the file COPYING that comes with GRASS
+#           for details.
+"""Create and display visualizations for space-time datasets."""
+
+import tempfile
+import os
+import weakref
+import grass.script as gs
+from .display import GrassRenderer
+
+
+def collect_lyr_dates(timeseries, etype):
+    """Create lists of layer names and start_times for a
+    space-time raster or vector dataset.
+
+    For datasets with variable timesteps, makes step regular with
+    "gran" method for t.rast.list or t.vect.list then fills in
+    missing layers with previous timestep layer. If first time step
+    is missing, uses the first available layer.
+
+    :param str timeseries: name of space-time dataset
+    :param str etype: element type, "stvds" or "strds"
+    """
+    if etype == "strds":
+        rows = gs.read_command(
+            "t.rast.list", method="gran", input=timeseries
+        ).splitlines()
+    elif etype == "stvds":
+        rows = gs.read_command(
+            "t.vect.list", method="gran", input=timeseries
+        ).splitlines()
+    else:
+        raise NameError(
+            _("Dataset {} must be element type 'strds' or 'stvds'").format(timeseries)
+        )
+
+    # Parse string
+    new_rows = [row.split("|") for row in rows]
+    new_array = [list(row) for row in zip(*new_rows)]
+
+    # Collect layer name and start time
+    for column in new_array:
+        if column[0] == "name":
+            names = column[1:]
+        if column[0] == "start_time":
+            dates = column[1:]
+
+    # For variable timestep datasets, fill in None values with
+    # previous time step value. If first time step is missing data,
```

Comment:
Could you give an example when the first will be None?

Reply:
I wasn't sure if this was possible so I wrote it that way out of an abundance of caution. I was thinking that since t.register lets you set a start datetime and increment, you might be able to have a space-time dataset with an empty first map.  But now, I'm doubting that it's possible to do that. @petrasovaa , are you aware of any way to create a dataset with None in the first step?